sdk/zksend/src/links.ts

```diff
@@ -33,24 +35,38 @@ const DEFAULT_ZK_SEND_LINK_OPTIONS = {
 	client: new SuiClient({ url: getFullnodeUrl('mainnet') }),
 };
 
-const SUI_COIN_TYPE = normalizeStructTag('0x2::coin::Coin<0x2::sui::SUI>');
+const SUI_COIN_TYPE = normalizeStructTag('0x2::sui::SUI');
+const SUI_COIN_OBJECT_TYPE = normalizeStructTag('0x2::coin::Coin<0x2::sui::SUI>');
+
+interface CreateZkSendLinkOptions {
+	calculateGas?: (options: {
+		mist: bigint;
+		objects: TransactionObjectInput[];
+		gasEstimateFromDryRun: bigint;
+	}) => Promise<bigint> | bigint;
+}
 
 export class ZkSendLinkBuilder {
 	#host: string;
 	#path: string;
 	#keypair: Keypair;
+	#client: SuiClient;
 	#objects = new Set<TransactionObjectInput>();
 	#mist = 0n;
-	#gasFee = 0n;
+	#fromAddress: string;
 
 	constructor({
 		host = DEFAULT_ZK_SEND_LINK_OPTIONS.host,
 		path = DEFAULT_ZK_SEND_LINK_OPTIONS.path,
 		keypair = new Ed25519Keypair(),
-	}: ZkSendLinkBuilderOptions = {}) {
+		client = DEFAULT_ZK_SEND_LINK_OPTIONS.client,
+		fromAddress,
+	}: ZkSendLinkBuilderOptions) {
 		this.#host = host;
 		this.#path = path;
 		this.#keypair = keypair;
+		this.#client = client;
+		this.#fromAddress = normalizeSuiAddress(fromAddress);
```

Comment:
should this be provided to `create` instead?

Reply:
I imagine you always need the sender because you need to dry run? Can we just rename this sender?

Reply:
we only dryRun during estimation which is now part of creating the send transaction, so thats where we need it...  Not sure it really matters either way